src/services/codefixes/codeFixProvider.ts

```diff
@@ -0,0 +1,50 @@
+﻿/* @internal */
+namespace ts {
+    export interface CodeFix {
+        errorCodes: string[];
+        getCodeActions(context: CodeFixContext): CodeAction[];
+    }
+
+    export interface CodeFixContext {
+        errorCode: string;
+        sourceFile: SourceFile;
+        span: TextSpan;
+        checker: TypeChecker;
+        newLineCharacter: string;
+    }
+
+    export namespace codefix {
+        const codeFixes = createMap<CodeFix[]>();
+
+        export function registerCodeFix(action: CodeFix) {
+            forEach(action.errorCodes, error => {
+                let fixes = codeFixes[error];
+                if (!fixes) {
+                    fixes = [];
+                    codeFixes[error] = fixes;
+                }
+                fixes.push(action);
+            });
+        }
+
+        export class CodeFixProvider {

```

Comment:
why do you need this class? all the state is in a const. so just loose the class, and export a `getCodeFixes` functions